Detectors/TPC/reconstruction/src/KrBoxClusterFinder.cxx

```diff
@@ -0,0 +1,250 @@
+    }
+    // here a "mMapOfAllDigits" large enough to contain all data points of the envents is
+    // initialized and the filled
+    mMapOfAllDigits = std::vector<std::vector<std::vector<int>>>(mMaxTimes, std::vector<std::vector<int>>(mMaxRows, std::vector<int>(mMaxPads)));
+    for (int i = 0; (cl + i) < eventSector->end(); i++) {
+      mMapOfAllDigits.at(Times.at(i)).at(Rows.at(i)).at(Pads.at(i)) = (cl + i)->getChargeFloat();
+    }
+  }
+}
+
+//#################################################
+
+// Function to update the temporal cluster
+void KrBoxClusterFinder::updateTempClusterFinal()
+{
+  double one_over_q_tot = 1. / mTempCluster.totCharge;
+  mTempCluster.meanPad *= one_over_q_tot;
+  mTempCluster.sigmaPad *= one_over_q_tot;
+  mTempCluster.meanRow *= one_over_q_tot;
+  mTempCluster.sigmaRow *= one_over_q_tot;
+  mTempCluster.meanTime *= one_over_q_tot;
+  mTempCluster.sigmaTime *= one_over_q_tot;
+  mTempCluster.sigmaPad = std::sqrt(std::abs(mTempCluster.sigmaPad - mTempCluster.meanPad * mTempCluster.meanPad));
+  mTempCluster.sigmaRow = std::sqrt(std::abs(mTempCluster.sigmaRow - mTempCluster.meanRow * mTempCluster.meanRow));
+  mTempCluster.sigmaTime = std::sqrt(std::abs(mTempCluster.sigmaTime - mTempCluster.meanTime * mTempCluster.meanTime));
+}
+
+// Function to update the temporal cluster.
+void KrBoxClusterFinder::updateTempCluster(int temp_Charge, int temp_Pad, int temp_Row, int temp_Time)
+{
+  if (temp_Charge >= mQThreshold) {
+    mTempCluster.size += 1;
+    mTempCluster.totCharge += temp_Charge;
+
+    mTempCluster.meanPad += temp_Pad * temp_Charge;
+    mTempCluster.sigmaPad += temp_Pad * temp_Pad * temp_Charge;
+
+    mTempCluster.meanRow += temp_Row * temp_Charge;
+    mTempCluster.sigmaRow += temp_Row * temp_Row * temp_Charge;
+
+    mTempCluster.meanTime += temp_Time * temp_Charge;
+    mTempCluster.sigmaTime += temp_Time * temp_Time * temp_Charge;
+    if (temp_Charge > mTempCluster.maxCharge) {
+      mTempCluster.maxCharge = temp_Charge;
+    }
+  } else {
+    std::cout << "Error: update cluster was called but current charge is below mQThreshold" << '\n';
+  }
+}
+
+// This function finds and evaluates all clusters in a 3D mMapOfAllDigits generated by the
+// mMapOfAllDigitsCreator function, this function also updates the cluster tree
+std::vector<std::tuple<int, int, int>> KrBoxClusterFinder::findLocalMaxima()
+{
+  std::vector<std::tuple<int, int, int>> localMaximaCoords;
+  // loop over whole mMapOfAllDigits the find clusers
+  for (int iRow = 0; iRow < mMaxRows; iRow++) {
+    for (int iPad = 0; iPad < mMaxPads; iPad++) {
+      for (int iTime = 0; iTime < mMaxTimes; iTime++) {
+
+        mTempCluster.totCharge = 0;
+        mTempCluster.maxCharge = 0;
+        mTempCluster.size = 0;
+        mTempCluster.meanPad = 0;
+        mTempCluster.meanRow = 0;
+        mTempCluster.meanTime = 0;
+        mTempCluster.sigmaPad = 0;
+        mTempCluster.sigmaRow = 0;
+        mTempCluster.sigmaTime = 0;

```

Comment:
it might be nicer if you add an inlined `reset` function to the KrCluster

Reply:
Done.